app/models/theme.rb

```diff
@@ -828,23 +836,44 @@ def scss_variables
     contents
   end
 
-  def convert_settings
-    settings.each do |setting|
-      setting_row = ThemeSetting.where(theme_id: self.id, name: setting.name.to_s).first
+  def migrate_settings
+    runner = ThemeSettingsMigrationsRunner.new(self)
+    results = runner.run
 
-      if setting_row && setting_row.data_type != setting.type
-        if (
-             setting_row.data_type == ThemeSetting.types[:list] &&
-               setting.type == ThemeSetting.types[:string] && setting.json_schema.present?
-           )
-          convert_list_to_json_schema(setting_row, setting)
-        else
-          Rails.logger.warn(
-            "Theme setting type has changed but cannot be converted. \n\n #{setting.inspect}",
-          )
-        end
+    return if results.blank?
+
+    old_settings = self.theme_settings.pluck(:name)
+    self.theme_settings.destroy_all
```

Comment:
Just curious why do we need to delete stuff first instead of just updating the existing value?